classes/slow_course_table.php

```diff
@@ -0,0 +1,162 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_excimer;
+
+/**
+ * Table for displaying grouped profile lists.
+ *
+ * @package   tool_excimer
+ * @author    Jason den Dulk <jasondendulk@catalyst-au.net>
+ * @copyright 2021, Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class slow_course_table extends profile_table {
+
+    /** Columns to be displayed. */
+    const COLUMNS = [
+        'maxduration',
+        'course',
+        'requestcount',
+        'maxcreated',
+        'mincreated',
+        'minduration',
+    ];
+
+    /** @var \moodle_url URL path to use for linking to profile groups. */
+    protected $urlpath;
+
+    /**
+     * Set the URL path to use for linking to profile groups.
+     *
+     * @param \moodle_url $url
+     */
+    public function set_url_path(\moodle_url $url) {
+        $this->urlpath = $url;
+    }
+
+    /**
+     * Sets the SQL for the table.
+     */
+    protected function put_sql(): void {
+        list($filterstring, $filterparams) = $this->get_filter_for_sql();
+
+        $this->set_count_sql(
+            "SELECT count(distinct request)
```

Comment:
Just for readability, all the SQL keywords should be caps eg distinct -> DISTINCT as well as the functions in the next query (MAX, COUNT) etc